// @google-cloud/sql/build/src/index.js
"use strict";
// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by synthtool. **
// ** https://github.com/googleapis/synthtool **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.protos = exports.SqlUsersServiceClient = exports.SqlTiersServiceClient = exports.SqlSslCertsServiceClient = exports.SqlOperationsServiceClient = exports.SqlInstancesServiceClient = exports.SqlFlagsServiceClient = exports.SqlDatabasesServiceClient = exports.SqlConnectServiceClient = exports.SqlBackupRunsServiceClient = exports.v1beta4 = exports.v1 = void 0;
const v1 = require("./v1");
exports.v1 = v1;
const v1beta4 = require("./v1beta4");
exports.v1beta4 = v1beta4;
const SqlBackupRunsServiceClient = v1.SqlBackupRunsServiceClient;
exports.SqlBackupRunsServiceClient = SqlBackupRunsServiceClient;
const SqlConnectServiceClient = v1.SqlConnectServiceClient;
exports.SqlConnectServiceClient = SqlConnectServiceClient;
const SqlDatabasesServiceClient = v1.SqlDatabasesServiceClient;
exports.SqlDatabasesServiceClient = SqlDatabasesServiceClient;
const SqlFlagsServiceClient = v1.SqlFlagsServiceClient;
exports.SqlFlagsServiceClient = SqlFlagsServiceClient;
const SqlInstancesServiceClient = v1.SqlInstancesServiceClient;
exports.SqlInstancesServiceClient = SqlInstancesServiceClient;
const SqlOperationsServiceClient = v1.SqlOperationsServiceClient;
exports.SqlOperationsServiceClient = SqlOperationsServiceClient;
const SqlSslCertsServiceClient = v1.SqlSslCertsServiceClient;
exports.SqlSslCertsServiceClient = SqlSslCertsServiceClient;
const SqlTiersServiceClient = v1.SqlTiersServiceClient;
exports.SqlTiersServiceClient = SqlTiersServiceClient;
const SqlUsersServiceClient = v1.SqlUsersServiceClient;
exports.SqlUsersServiceClient = SqlUsersServiceClient;
exports.default = {
    v1,
    v1beta4,
    SqlBackupRunsServiceClient,
    SqlConnectServiceClient,
    SqlDatabasesServiceClient,
    SqlFlagsServiceClient,
    SqlInstancesServiceClient,
    SqlOperationsServiceClient,
    SqlSslCertsServiceClient,
    SqlTiersServiceClient,
    SqlUsersServiceClient,
};
const protos = require("../protos/protos");
exports.protos = protos;
