// @google-cloud/sql/build/src/v1beta4/index.js
"use strict";
// Copyright 2024 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// ** This file is automatically generated by gapic-generator-typescript. **
// ** https://github.com/googleapis/gapic-generator-typescript **
// ** All changes to this file may be overwritten. **
Object.defineProperty(exports, "__esModule", { value: true });
exports.SqlUsersServiceClient = exports.SqlTiersServiceClient = exports.SqlSslCertsServiceClient = exports.SqlOperationsServiceClient = exports.SqlInstancesServiceClient = exports.SqlFlagsServiceClient = exports.SqlDatabasesServiceClient = exports.SqlConnectServiceClient = exports.SqlBackupRunsServiceClient = void 0;
var sql_backup_runs_service_client_1 = require("./sql_backup_runs_service_client");
Object.defineProperty(exports, "SqlBackupRunsServiceClient", { enumerable: true, get: function () { return sql_backup_runs_service_client_1.SqlBackupRunsServiceClient; } });
var sql_connect_service_client_1 = require("./sql_connect_service_client");
Object.defineProperty(exports, "SqlConnectServiceClient", { enumerable: true, get: function () { return sql_connect_service_client_1.SqlConnectServiceClient; } });
var sql_databases_service_client_1 = require("./sql_databases_service_client");
Object.defineProperty(exports, "SqlDatabasesServiceClient", { enumerable: true, get: function () { return sql_databases_service_client_1.SqlDatabasesServiceClient; } });
var sql_flags_service_client_1 = require("./sql_flags_service_client");
Object.defineProperty(exports, "SqlFlagsServiceClient", { enumerable: true, get: function () { return sql_flags_service_client_1.SqlFlagsServiceClient; } });
var sql_instances_service_client_1 = require("./sql_instances_service_client");
Object.defineProperty(exports, "SqlInstancesServiceClient", { enumerable: true, get: function () { return sql_instances_service_client_1.SqlInstancesServiceClient; } });
var sql_operations_service_client_1 = require("./sql_operations_service_client");
Object.defineProperty(exports, "SqlOperationsServiceClient", { enumerable: true, get: function () { return sql_operations_service_client_1.SqlOperationsServiceClient; } });
var sql_ssl_certs_service_client_1 = require("./sql_ssl_certs_service_client");
Object.defineProperty(exports, "SqlSslCertsServiceClient", { enumerable: true, get: function () { return sql_ssl_certs_service_client_1.SqlSslCertsServiceClient; } });
var sql_tiers_service_client_1 = require("./sql_tiers_service_client");
Object.defineProperty(exports, "SqlTiersServiceClient", { enumerable: true, get: function () { return sql_tiers_service_client_1.SqlTiersServiceClient; } });
var sql_users_service_client_1 = require("./sql_users_service_client");
Object.defineProperty(exports, "SqlUsersServiceClient", { enumerable: true, get: function () { return sql_users_service_client_1.SqlUsersServiceClient; } });
